querybook/webapp/lib/utils/complex-types.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Convert a complex Hive type string to a nested JSON object
+ *
+ * Example: 'struct<date:struct<year:int,month:int,day:int>,hour:int,minute:int,second:int,timeZoneId:string>'
+ * Output: {
+        date: {
+            year: 'int',
+            month: 'int',
+            day: 'int',
+        },
+        hour: 'int',
+        minute: 'int',
+        second: 'int',
+        timeZoneId: 'string',
+    }
+ */
+export function parseType(type: string): any | string {
```

Comment:
will there be cases of 
* uppercase, e.g. `STRUCT`
* space between, e.g. `struct <`

and what happens if the parsing fails?